src/main/java/frc/robot/subsystems/snapback/shooter/ShooterIOSim.java

```diff
@@ -1,3 +1,110 @@
 package frc.robot.subsystems.snapback.shooter;
 
-public class ShooterIOSim {}
+import edu.wpi.first.math.MathUtil;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.system.plant.LinearSystemId;
+import edu.wpi.first.wpilibj.simulation.DCMotorSim;
+import frc.robot.constants.Constants;
+
+public class ShooterIOSim implements ShooterIO {
+
+  private DCMotorSim leftFlywheelMotorSim;
+  private DCMotorSim rightFlywheelMotorSim;
+  private DCMotorSim acceleratorMotorSim;
+
+  private double leftFlywheelMotorAppliedVolts;
+  private double rightFlywheelMotorAppliedVolts;
+  private double acceleratorMotorAppliedVolts;
+
```

Comment:
Because there isn't actual hardware in the simulated instance of the shooter, we need to create a motion profile and a PID controller for the simulated hardware.


```suggestion
private ProfiledPIDController leftController;
private ProfiledPIDController rightController;
```

Check the WPIlib documentation for more information about how this class works!